app/federation/server/hooks/afterRemoveFromRoom.js

```diff
@@ -0,0 +1,53 @@
+import { FederationRoomEvents } from '../../../models/server';
+import { isFederated, getFederatedRoomData } from '../functions/helpers';
+import { logger } from '../lib/logger';
+import { normalizers } from '../normalizers';
+import { getFederationDomain } from '../lib/getFederationDomain';
+import { dispatchEvent } from '../handler';
+
+async function afterRemoveFromRoom(involvedUsers, room) {
+	const { removedUser } = involvedUsers;
+
+	// If there are not federated users on this room, ignore it
+	if (!isFederated(room) && !isFederated(removedUser)) { return; }
+
+	logger.client.debug(() => `afterRemoveFromRoom => involvedUsers=${ JSON.stringify(involvedUsers, null, 2) } room=${ JSON.stringify(room, null, 2) }`);
+
+	const { users } = getFederatedRoomData(room);
+
+	try {
+		// Get the domains after removal
+		const domainsAfterRemoval = users.map((u) => u.federation.origin);
+
+		//
+		// Normalize the room's federation status
+		//
+		const usersBeforeRemoval = users;
+		usersBeforeRemoval.push(removedUser);
+
+		// Get the users domains
+		const domainsBeforeRemoval = usersBeforeRemoval.map((u) => u.federation.origin);
+
+		//
+		// Create the user remove event
+		//
+		const normalizedSourceUser = normalizers.normalizeUser(removedUser);
+
+		const removeUserEvent = await FederationRoomEvents.createRemoveUserEvent(getFederationDomain(), room._id, normalizedSourceUser, domainsAfterRemoval);
+
+		// Dispatch the events
+		dispatchEvent(domainsBeforeRemoval, removeUserEvent);
+	} catch (err) {
+		logger.client.error(() => `afterRemoveFromRoom => involvedUsers=${ JSON.stringify(involvedUsers, null, 2) } => Could not add user: ${ err }`);
+
+		throw err;
```

Comment:
I dont think we should throw on `after` events... because we gonna break the callbacks